src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -157,6 +169,185 @@ namespace OpenMS
+      {
+        if (peak_nr == 0 && it != chromatogram.begin())
+        {
+          // add area between first measured peak inside the peak boundaries and start of peak to the left
+          double delta_rt = it->getRT() - best_left;
+          double interpol_intensity = lin_interpolate(best_left, prev_it->getRT(), it->getRT(), prev_it->getIntensity(), it->getIntensity());
+          intensity_integral += (interpol_intensity + it->getIntensity())/2.0 * delta_rt;
+        }
+
+        if (peak_nr > 0)
+        {
+          // add area between last peak (inside boundaries) and current peak (inside boundaries)
+          double delta_rt = it->getRT() - prev_it->getRT();
+          intensity_integral += (prev_it->getIntensity() + it->getIntensity())/2.0 * delta_rt;
+        }
+
+        DPosition<2> p;
+        p[0] = it->getMZ();
+        p[1] = it->getIntensity();
+        hull_points.push_back(p);
+        if (std::fabs(it->getMZ() - peak_apex_rt) <= peak_apex_dist)
+        {
+          peak_apex_int = p[1];
+          peak_apex_dist = std::fabs(it->getMZ() - peak_apex_rt);
+        }
+        rt_sum += it->getMZ();
+        intensity_sum += it->getIntensity();
+        peak_nr++;
+      }
+      else if (peak_nr > 0 && !added_right)
+      {
+        // add area between last measured peak inside the peak boundaries and end of peak to the right
+        double delta_rt = best_right - prev_it->getRT();
+        double interpol_intensity = lin_interpolate(best_right, prev_it->getRT(), it->getRT(), prev_it->getIntensity(), it->getIntensity());
+        intensity_integral += (prev_it->getIntensity() + interpol_intensity)/2.0 * delta_rt;
+        added_right = true;
+      }
+      prev_it = it;
+    }
+  }
+
+  void MRMTransitionGroupPicker::calculatePeakShapeMetrics_(const MSChromatogram& chromatogram, 
+    double best_left, double best_right, 
+    double peak_height, double peak_apex_rt, double avg_noise_level,
+    PeakShapeMetrics_ & peakShapeMetrics)
+  {
+    peakShapeMetrics.points_across_baseline = 0;
+    double start_intensity(0), end_intensity(0);
+    double delta_rt, delta_int, height_5, height_10, height_50;
+    
+    // pass 1
+    for (MSChromatogram::const_iterator it = chromatogram.begin() + 1; it != chromatogram.end(); ++it)
+    {
+      MSChromatogram::const_iterator it_prev = it;
+      --it_prev; //previous point
+      double intensity = std::max(it->getIntensity()-avg_noise_level, 0.0); //background-subtracted intensity
+      double intensity_prev = std::max(it_prev->getIntensity()-avg_noise_level, 0.0); //background-subtracted intensity of the previous point
+      double retention_time = it->getMZ();
+      double retention_time_prev = it_prev->getMZ();
+
+      // start and end intensities
+      if (retention_time_prev <= best_left && retention_time >= best_left)
+      {
+        start_intensity = intensity;
+      }
+      else if (retention_time_prev <= best_right && retention_time >= best_right)
```

Comment:
hmm now thinking about this, it feels that its still not quite correct. I think it should be correct like this:

```
retention_time_prev < best_right && retention_time >= best_right
```

otherwise, this condition will trigger twice in the case where `best_right` is exactly at a RT position, once when it is equal to the previous one and once when it is equal to the current one. Same for above.

Reply:
I see what you mean.  That is probably why the tests changed also.  I will fix that.